File: Week 0/Reverse String/reverse_string.swift
 /* 
 Challenge: https://leetcode.com/problems/reverse-string/
 Explain method swap in Swift: https://developer.apple.com/documentation/swift/array/2893281-swapat
*/

class Solution {
    func reverseString(_ s: inout [Character]) {
        var i = 0
        var j = s.count-1
        
        while i < j {
            s.swapAt(i,j)
            i += 1
            j -= 1
        }
    }
}

File: Week 0/Reverse String/reverse_string.java
/* Challenge: https://leetcode.com/problems/reverse-string/
 
Explain method swap in Java https://www.geeksforgeeks.org/collections-swap-method-in-java-with-examples/
*/

class Solution {
  public void reverseString(char[] s) {
    int i = 0;
    int j = s.length - 1;
    while (i < j) {
      char temp = s[i];
      s[i] = s[j];
      s[j] = temp;
      i++;
      j--;
    }
  }
}
